components/organisation-unit-tree/src/organisation-unit-tree/default-render-node-label/label/label.js

```diff
@@ -0,0 +1,134 @@
+import { SingleSelectionLabel } from './single-selection-label.js'
+
+const createNewSelected = ({ selected, path, checked, singleSelection }) => {
+    const pathIndex = selected.indexOf(path)
+
+    if (checked && pathIndex !== -1) return selected
+    if (singleSelection && checked) return [path]
+    if (checked) return [...selected, path]
+    if (pathIndex === -1) return selected
+    if (singleSelection) return []
+    if (selected.indexOf(path) === 0) return selected.slice(1)
+
+    const prevSlice = selected.slice(0, pathIndex)
+    const nextSlice = selected.slice(pathIndex + 1)
+    return [...prevSlice, ...nextSlice]
+}
+
+const Label = ({
+    children,
+    node,
+    open,
+    loading,
+    checked,
+    onChange,
+    dataTest,
+    selected,
+    hasChildren,
+    highlighted,
+    onToggleOpen,
+    disableSelection,
+    singleSelection,
+    hasSelectedDescendants,
+}) => {
+    const onClick = ({ checked }, event) => {
+        const newSelected = createNewSelected({
+            path: node.path,
+            selected,
+            checked,
+            singleSelection,
+        })
+
+        onChange(
+            {
+                // @TODO: It'd make more sense to pass the node as an object
+                // isntead of spread it. But that'd be a breaking change
+                ...node,
+                checked,
+                selected: newSelected,
+            },
+            event
+        )
+    }
+
+    if (disableSelection) {
+        return (
+            <LabelContainer highlighted={highlighted}>
+                <DisabledSelectionLabel
+                    loading={loading}
+                    onToggleOpen={onToggleOpen}
+                >
+                    {children}
+                </DisabledSelectionLabel>
+            </LabelContainer>
+        )
+    }
+
+    if (singleSelection) {
+        return (
+            <LabelContainer highlighted={highlighted}>
+                <SingleSelectionLabel
+                    checked={checked}
+                    onChange={onClick}
+                    loading={loading}
+                >
+                    {children}
+                </SingleSelectionLabel>
+            </LabelContainer>
+        )
+    }
+
+    return (
+        <LabelContainer highlighted={highlighted}>
+            <IconizedCheckbox
+                dataTest={dataTest}
+                checked={checked}
+                name="org-unit"
+                value={node.id}
+                loading={loading}
+                indeterminate={!checked && hasSelectedDescendants}
+                onChange={onClick}
+                open={open}
+                hasChildren={hasChildren}
+            >
+                {children}
+            </IconizedCheckbox>
+        </LabelContainer>
+    )
+}
+
+Label.propTypes = {
+    children: propTypes.any.isRequired,
+    dataTest: propTypes.string.isRequired,
+    // This is `any` so it can be customized by the app
```

Comment:
looks like the `propType` that this comment is associated with was moved (by eslint?) but the comment was not moved with it